src/main/java/com/launchdarkly/eventsource/EventSource.java

```diff
@@ -23,18 +20,18 @@
 
   private final URI uri;
   private final Headers headers;
-  private final ExecutorService executor;
+  private final ExecutorService executor = Executors.newCachedThreadPool();
```

Comment:
I think this is less clear, because all of the other class members are initialized in the ctor.